pacsman/dicom_interface.py

```diff
@@ -132,3 +142,30 @@ def fetch_thumbnail(self, series_id):
         :return: A path to a PNG file on success, None if not found
         """
         raise NotImplementedError
+
+    @staticmethod
+    def build_patient_result(result, ds, additional_tags=None):
+        patient_id = getattr_required(ds, 'PatientID')
+        study_instance_uid = getattr_required(ds, 'StudyInstanceUID')
+
+        if len(result) == 0:
+            result.PatientID = patient_id
+            result.PatientName = getattr(ds, 'PatientName', '')
+            result.PatientBirthDate = getattr(ds, 'PatientBirthDate', '')
+            result.PatientStudyInstanceUIDs = MultiValue(UID, [study_instance_uid])
+            result.PacsmanPrivateIdentifier = PRIVATE_ID
+            result.PatientMostRecentStudyDate = getattr(ds, 'StudyDate', '')
+            copy_dicom_attributes(result, ds, additional_tags)
+        else:
+            if result.PatientID != patient_id:
+                raise ValueError(f"The search result has a different patient ID")
+
+            existing_uids = {uid.name for uid in result.PatientStudyInstanceUIDs}
```

Comment:
@dillonwilliams one of the downsides of using the "UI" type is that we have to grab `uid.name` in order to do these types of comparisons.